src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -520,6 +520,15 @@ std::string ov::npuw::CompiledModel::funcall_mem_device(const std::size_t idx) c
     //        It's not feasible to allocate them in L0 due to high memory consumption.
     //        Until we make such memory reusable, hard-coding those tensors to CPU.
     return "CPU";
+
+    // Force globally set device if set
+    const std::string device_alloc = m_cfg.get<::intel_npu::NPUW_WEIGHTS_BANK_ALLOC>();
+    if (!device_alloc.empty()) {
+        return device_alloc;
+    }
+
+    auto& comp_model_desc = m_compiled_submodels[idx];
+    return *comp_model_desc.device_it;
```

Comment:
Static analysis will for sure fire unreachable code here, so the past body should've stayed under `#if 0`. But let's see if it works